File: tdf-hq/src/TDF/API/Rooms.hs
{-# LANGUAGE DataKinds #-}
{-# LANGUAGE TypeOperators #-}

module TDF.API.Rooms where

import           Data.Text (Text)
import           Servant

import           TDF.API.Types

type RoomsPublicAPI =
       "rooms" :> "public" :> Get '[JSON] [RoomDTO]

type RoomsAPI =
       "rooms" :> Get '[JSON] [RoomDTO]
  :<|> "rooms" :> ReqBody '[JSON] RoomCreate :> PostCreated '[JSON] RoomDTO
  :<|> "rooms" :> Capture "id" Text :> ReqBody '[JSON] RoomUpdate :> Patch '[JSON] RoomDTO

File: archives/tdf-root-ui-legacy/api/rooms.ts
import { get, post, patch } from './client';
import type { RoomCreate, RoomDTO, RoomUpdate } from './types';

export const Rooms = {
  list: () => get<RoomDTO[]>('/rooms'),
  create: (body: RoomCreate) => post<RoomDTO>('/rooms', body),
  update: (id: string, body: RoomUpdate) => patch<RoomDTO>(`/rooms/${id}`, body),
};
